packages/@aws-cdk/aws-iam/lib/private/imported-role.ts

```diff
@@ -43,7 +45,11 @@ export class ImportedRole extends Resource implements IRole, IComparablePrincipa
 
   public addToPrincipalPolicy(statement: PolicyStatement): AddToPrincipalPolicyResult {
     if (!this.defaultPolicy) {
-      this.defaultPolicy = new Policy(this, this.defaultPolicyName ?? 'Policy');
+      const defaultDefaultPolicyName = FeatureFlags.of(this).isEnabled(IAM_IMPORTED_ROLE_STACK_SAFE_DEFAULT_POLICY_NAME)
+          ? `Policy${Names.uniqueId(this)}`
+          : 'Policy';
+      const policyName = this.defaultPolicyName ?? defaultDefaultPolicyName;
+      this.defaultPolicy = new Policy(this, policyName, { policyName });
```

Comment:
Pretty sure this still changes the existing behavior when a `defaultPolicyName` is provided as well as the Feature Flag being disabled.

You'll want `policyName` to be only set if no `defaultPolicyName` is set and the FF is enabled.

Reply:
Good catch!

Reply:
I've just fixed that.